activity/kafkapub/activity.go

```diff
@@ -0,0 +1,171 @@
+}
+
+// Metadata implements activity.Activity.Metadata
+func (a *KafkaPubActivity) Metadata() *activity.Metadata {
+	return a.metadata
+}
+
+// Eval implements activity.Activity.Eval
+func (a *KafkaPubActivity) Eval(context activity.Context) (done bool, err error) {
+	err = initParms(a, context)
+	if err != nil {
+		flogoLogger.Errorf("Kafkapub parameters initialization got error: [%s]", err)
+		return false, err
+	}
+	defer func() {
+		if err := a.syncProducer.Close(); err != nil {
+			panic(err)
+		}
+	}()
+	if context.GetInput("Message") != nil {
+		msg := &sarama.ProducerMessage{
+			Topic: a.topic,
+			Value: sarama.StringEncoder(context.GetInput("Message").(string)),
+		}
+		partition, offset, err := a.syncProducer.SendMessage(msg)
+		if err != nil {
+			return false, fmt.Errorf("kafkapub failed to send message for reason [%s]", err)
+		}
+		context.SetOutput("partition", partition)
+		context.SetOutput("offset", offset)
+		flogoLogger.Debugf("Kafkapub message [%s] sent successfully on partition [%d] and offset [%d]",
+			context.GetInput("Message").(string), partition, offset)
+		return true, nil
+	}
+	return false, fmt.Errorf("kafkapub called without a message to publish")
+}
+
+func initParms(a *KafkaPubActivity, context activity.Context) error {
+	if context.GetInput("BrokerUrls") != nil {
+		a.kafkaConfig = sarama.NewConfig()
+		a.kafkaConfig.Producer.Return.Errors = true
+		a.kafkaConfig.Producer.RequiredAcks = sarama.WaitForAll
+		a.kafkaConfig.Producer.Retry.Max = 5
+		a.kafkaConfig.Producer.Return.Successes = true
+		brokerUrls := strings.Split(context.GetInput("BrokerUrls").(string), ",")
+		brokers := make([]string, len(brokerUrls))
+		for brokerNo, broker := range brokerUrls {
+			_, error := url.Parse(broker)
+			if error != nil {
+				return fmt.Errorf("BrokerUrl [%s] format invalid for reason: [%s]", broker, error.Error())
+			}
+			brokers[brokerNo] = broker
+		}
+		a.brokers = brokers
+		flogoLogger.Debugf("Kafkapub brokers [%v]", brokers)
+	} else {
+		return fmt.Errorf("Kafkapub activity is not configured with at least one BrokerUrl")
+	}
+	if context.GetInput("Topic") != nil {
+		a.topic = context.GetInput("Topic").(string)
+		flogoLogger.Debugf("Kafkapub topic [%s]", a.topic)
+	} else {
+		return fmt.Errorf("Topic input parameter not provided and is required")
+	}
+
+	//clientKeystore
+	/*
+		Its worth mentioning here that when the keystore for kafka is created it must support RSA keys via
+		the -keyalg RSA option.  If not then there will be ZERO overlap in supported cipher suites with java.
+		see:   https://issues.apache.org/jira/browse/KAFKA-3647
+		for more info
+	*/
+	if context.GetInput("truststore") != nil {
+		trustStore := context.GetInput("truststore")
+		if trustStore != nil && len(trustStore.(string)) > 0 {
+			trustPool, err := getCerts(trustStore.(string))
+			if err != nil {
+				return err
+			}
+			config := tls.Config{
+				RootCAs:            trustPool,
+				InsecureSkipVerify: true}
```

Comment:
Looks funny, but go fmt did not change it.  I moved the closing brace down a line.